microcosm_pubsub/tests/test_consumer.py

```diff
@@ -119,26 +125,27 @@ def test_json_consume():
     Test that message sent as JSON could be delivered
 
     """
-    graph = create_daemon_with_naive_cache()
-    # simulate the response structure
-    # replacing MagicMock with real reader
-    graph.sqs_consumer.sqs_client = SQSJsonReader(dict(
+    event = dict(
             MessageId=MESSAGE_ID,
             ReceiptHandle=RECEIPT_HANDLE,
             MD5OfBody="7efaa8404863d47c51ed0e20b9014aec",
             Body=dumps(dict(
                 data="data",
                 mediaType=DerivedSchema.MEDIA_TYPE,
             )),
-        ),
-    )
+        )
 
+    graph = create_daemon_with_naive_cache(event)
+    # simulate the response structure
+    # replacing MagicMock with real reader
+    graph.sqs_consumer.sqs_client = SQSJsonReader(event)
     messages = graph.sqs_consumer.consume()
 
     # and response translated properly
+    assert_that(graph.sqs_consumer, is_(instance_of(SQSConsumer)))
+    assert_that(graph.sqs_envelope, is_(instance_of(CodecSQSEnvelope)))
     assert_that(messages, has_length(1))
     assert_that(messages[0].consumer, is_(equal_to(graph.sqs_consumer)))
```

Comment:
Isn't there another assertion you could add checking that the config is as expected? IIRC when instantiating the daemon you're adding config for the Json case.